.github/workflows/swift.yml

```diff
@@ -0,0 +1,28 @@
+name: Swift
+
+on: 
+    push:
+        branches:
+            - main
+    pull_request:
+        branches:
+            - main
```

Comment:
Can we limit this to running only if we have changes to the Swift code? If I understand the macos runner pricing correctly, 13 minutes of CI runtime (this job) would cost us a dollar on each run of this job. We are running the CI a lot, and I want to avoid unnecessary costs.

Let me know if I am mistaken somewhere.

Reply:
I think it is already set to only run when swift code changes (line 15)

Reply:
@danielnugraha, the line with `working-directory` just defines the from where the commands are run from. 

You can use this to limit the directory in which the change occurred:
* https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-paths